src/clj_irods/core.clj

```diff
@@ -109,20 +119,23 @@
             listing-item (when (delay? paged-folder-listing)
                            (get-from-listing (apply concat (icat/flatten-cached-listings @paged-folder-listing))))]
         (or
-          (when listing-item (delay (extract-fn listing-item)))
-          (when single-item  (delay (extract-fn @single-item)))
+          (when listing-item (instrumented-delay (extract-fn listing-item)))
+          (when single-item  (instrumented-delay (extract-fn @single-item)))
           nil))
       ;; better if we pass user group IDs in so we can get them from any source, rather than forcing the icat version specifically
       (when (:has-icat irods)
-        (let [userids (icat/user-group-ids irods user zone)]
-          (delay (force userids) (extract-fn @(icat/get-item irods user zone path))))))))
+        (let [userids (icat/user-group-ids irods user zone)
+              item (future (force userids) @(icat/get-item irods user zone path))]
+          (instrumented-delay (extract-fn @item)))))))
```

Comment:
This is a somewhat funny thing. How I'd written it before, it wouldn't start trying to get the item until the delay returned here was deref'd, but the usual contract is that it starts getting it right away, but it only reads it when the delay is read. The use of `future` makes sure that happens while still letting us force the `userids` delay to ensure those are cached before getting the item.